libraries/resource_providers/api200/volume_template_provider.rb

```diff
@@ -15,26 +15,33 @@ module API200
     class VolumeTemplateProvider < ResourceProvider
       # Loads the VolumeTemplate with all the external resources (if needed)
       def load_resource_with_associated_resources
-        raise "Unspecified property: 'storage_system'. Please set it before attempting this action." unless @new_resource.storage_system
-        raise "Unspecified property: 'storage_pool'. Please set it before attempting this action." unless @new_resource.storage_pool
         @item['provisioning']['capacity'] = @item['provisioning']['capacity'].to_s if @item['provisioning'] && @item['provisioning']['capacity']
-        load_storage_system
-        sp = resource_named(:StoragePool).find_by(@item.client, name: @new_resource.storage_pool, storageSystemUri: @item['storageSystemUri']).first
-        raise "Storage Pool '#{@new_resource.storage_pool}' not found for Storage System '#{@new_resource.storage_system}'" unless sp
-        @item['provisioning']['storagePoolUri'] = sp['uri']
+        @item.set_storage_system(load_storage_system)
+        @item['provisioning']['storagePoolUri'] = load_storage_pool['uri']
         @item.set_snapshot_pool(resource_named(:StoragePool).new(@item.client, name: @new_resource.snapshot_pool)) if @new_resource.snapshot_pool
       end
 
-      # Loads Storage System in the given VolumeTemplate resource.
+      # Loads the Storage Pool
+      # @raise [RuntimeError] if property 'storage_pool' is not set or Storage Pool resource not found
+      # @return [resource_named(:StoragePool)] The Storage Pool loaded
+      def load_storage_pool
+        load_storage_system unless @storage_system
+        raise "Unspecified property: 'storage_pool'. Please set it before attempting this action." unless @new_resource.storage_pool
+        storage_pool = resource_named(:StoragePool).find_by(@item.client, name: @new_resource.storage_pool, storageSystemUri: @storage_system['uri']).first
+        storage_pool || raise("Storage Pool '#{@new_resource.storage_pool}' not found for Storage System '#{@new_resource.storage_system}'")
+      end
+
+      # Loads the Storage System
       # The property storage_system needs to be used in the recipe for this code to load the Storage System.
       # Hostname or storage system name can be used
-      # @return [resource_named(:VolumeTemplate)] VolumeTemplate with Storage System parameters updated
+      # @return [resource_named(:StorageSystem)] The StorageSystem loaded
       def load_storage_system
```

Comment:
This whole method feels kind of unecessary. `load_resource` accepts both the `name` or a `hash`, why not simply specify in the documentation that @new_resource.storage_system can be either of those and drop this entirely?